apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/create-flow/SharedWalletCreationStore.tsx

```diff
@@ -73,7 +74,15 @@ export type StateQuorum = MakeState<{
   walletName: string;
 }>;
 
-type State = StateSetup | StateCoSigners | StateCoSignersImportantInfo | StateQuorum;
+export type StateShareDetails = MakeState<
+  StateCoSignersCommon & {
+    coSignersKeys: string[];
+    step: SharedWalletCreationStep.ShareDetails;
+    walletName: string;
+  }
+>;
```

Comment:
The `StateCoSignersCommon` type is related to the CoSigners step. I would rather avoid reusing it here. It is there because we have two CoSigngers steps (the second one displays modal) and both are identical except for the step property. Also, there is no `coSignersKeys` property.

However, if you strongly want to extend some other type I would suggest extending the quorum state without the step property. This might make sense because the state doesn't change between the quorum and the share details step.
```suggestion
export type StateShareDetails = MakeState<
  Omit<StateQuorum, 'step'> & {
    step: SharedWalletCreationStep.ShareDetails;
  }
>;
```